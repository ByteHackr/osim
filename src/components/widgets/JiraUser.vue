<script setup lang="ts">
import { computed, toValue } from 'vue';

const props = defineProps<{
  displayName: string;
  emailAddress?: string;
  name?: string;
  query?: string;
}>();

const formatedUser = computed(() => {
  let output = props.displayName;
  if (props.emailAddress) {
    output += ` - ${props.emailAddress}`;
  }
  if (props.name) {
    output += ` (${props.name})`;
  }
  if (props.query) {
    output = output.replace(new RegExp(toValue(props.query), 'gi'), '<b>$&</b>');
  }
  return output;
});
</script>

<template>
  <!-- eslint-disable-next-line vue/no-v-html -->
  <span v-html="formatedUser" />
</template>
