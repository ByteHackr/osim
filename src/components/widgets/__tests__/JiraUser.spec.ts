import { mount } from '@vue/test-utils';

import JiraUser from '../JiraUser.vue';

describe('jiraUser', () => {
  it('renders correctly with only displayName', () => {
    const wrapper = mount(JiraUser, {
      props: { displayName: 'John Doe' },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with displayName and emailAddress', () => {
    const wrapper = mount(JiraUser, {
      props: {
        displayName: 'John Doe',
        emailAddress: 'john.doe@example.com',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('renders correctly with all props', () => {
    const wrapper = mount(JiraUser, {
      props: {
        displayName: 'John Doe',
        emailAddress: 'john.doe@example.com',
        name: 'johndoe',
        query: 'Doe',
      },
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('highlights the query text', () => {
    const wrapper = mount(JiraUser, {
      props: {
        displayName: 'John Doe',
        query: 'Doe',
      },
    });
    expect(wrapper.html()).toContain('<b>Doe</b>');
    expect(wrapper.html()).toMatchSnapshot();
  });
});
