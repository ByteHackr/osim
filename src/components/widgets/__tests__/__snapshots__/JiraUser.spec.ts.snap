// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`JiraUser > highlights the query text 1`] = `
"<!-- eslint-disable-next-line vue/no-v-html -->
<span>John <b>Doe</b></span>"
`;

exports[`JiraUser > renders correctly with all props 1`] = `
"<!-- eslint-disable-next-line vue/no-v-html -->
<span>John <b>Doe</b> - john.<b>doe</b>@example.com (john<b>doe</b>)</span>"
`;

exports[`JiraUser > renders correctly with displayName and emailAddress 1`] = `
"<!-- eslint-disable-next-line vue/no-v-html -->
<span>John Doe - john.doe@example.com</span>"
`;

exports[`JiraUser > renders correctly with only displayName 1`] = `
"<!-- eslint-disable-next-line vue/no-v-html -->
<span>John Doe</span>"
`;
